src/main/java/org/folio/linked/data/service/resource/AssignAuthorityTarget.java

```diff
@@ -0,0 +1,45 @@
+package org.folio.linked.data.service.resource;
+
+import static java.lang.Boolean.FALSE;
+import static java.util.Map.of;
+import static org.apache.commons.lang3.ObjectUtils.notEqual;
+import static org.folio.ld.dictionary.ResourceTypeDictionary.CONCEPT;
+import static org.folio.ld.dictionary.ResourceTypeDictionary.FAMILY;
+import static org.folio.ld.dictionary.ResourceTypeDictionary.JURISDICTION;
+import static org.folio.ld.dictionary.ResourceTypeDictionary.MEETING;
+import static org.folio.ld.dictionary.ResourceTypeDictionary.ORGANIZATION;
+import static org.folio.ld.dictionary.ResourceTypeDictionary.PERSON;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.Set;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import org.apache.commons.collections4.CollectionUtils;
+import org.folio.ld.dictionary.ResourceTypeDictionary;
+
+@Getter
+@RequiredArgsConstructor
+public enum AssignAuthorityTarget {
+
+  CREATOR_OF_WORK(of(CONCEPT, Set.of(PERSON, FAMILY, ORGANIZATION, JURISDICTION, MEETING)));
+
+  private final Map<ResourceTypeDictionary, Set<ResourceTypeDictionary>> combinedTypes;
+
+  public Boolean isCompatibleWith(Collection<ResourceTypeDictionary> types) {
+    if (CollectionUtils.isEmpty(types)) {
+      return FALSE;
+    }
+    if (types.size() == 1) {
+      return combinedTypes.values().stream().flatMap(Collection::stream)
+        .anyMatch(types::contains);
+    }
+    return combinedTypes.entrySet()
```

Comment:
This logic is not exactly correct.
With this logic, resource of type "Concept, Person" will return true for "CREATOR_OF_WORK". We have to adjust logic so that true will be returned only if Type of the resource is exactly Person or Meeting or Jurisdiction or Organization or Family.

For SUBJECT_OF_WORK (future story), true should be returned for both "Person" or "Concept, Person".

A real life example:

Let us consider two resources:
1. "Barack Obama" -> This will be a resource of type "Person"
2. "Presidency of Barack Obama" -> This will be a resource of type "Concept, Person"

You can assign resource 1 as creator of a work. However, it doesn't make sense to assign resource 2 as creator of a work.

Both resources can be assigned as the subject of a work. (ie, someone can write a book about "Barack Obama" or the "Presidency of Barack Obama". Both are valid.)

Here is a DRAFT implementation of this logic. Pleas take a look.

```
public enum AssignAuthorityTarget {

  CREATOR_OF_WORK(Set.of(PERSON, FAMILY, ORGANIZATION, JURISDICTION, MEETING), Set.of()),
  SUBJECT_OF_WORK(Set.of(FAMILY, ORGANIZATION, JURISDICTION, MEETING, PERSON, FORM, TOPIC, TEMPORAL, PLACE),
    Set.of(CONCEPT));

  private final Set<ResourceTypeDictionary> exact;
  private final Set<ResourceTypeDictionary> any;

  public Boolean isCompatibleWith(Collection<ResourceTypeDictionary> types) {
    if (types.size() == 1 && exact.containsAll(types)) {
      return true;
    }
    return CollectionUtils.containsAny(types, any);
  }
}
```

Reply:
thank you PK

Reply:
Done.